test/core.test.js

```diff
@@ -0,0 +1,524 @@
+    });
+
+    it("Should deploy Auth connector", async function () {
+      await deployConnector({
+        connectorName: "authV2",
+        contract: "ConnectV2Auth",
+        abi: (await deployments.getArtifact("ConnectV2Auth")).abi
+      })
+      expect(!!addresses.connectors["authV2"]).to.be.true
+      await instaConnectorsV2.connect(masterSigner).toggleConnectors([addresses.connectors["authV2"]])
+    });
+
+    it("Should deploy EmitEvent connector", async function () {
+      await deployConnector({
+        connectorName: "emitEvent",
+        contract: "ConnectV2EmitEvent",
+        abi: (await deployments.getArtifact("ConnectV2EmitEvent")).abi
+      })
+      expect(!!addresses.connectors["emitEvent"]).to.be.true
+      await instaConnectorsV2.connect(masterSigner).toggleConnectors([addresses.connectors["emitEvent"]])
+    });
+
+    it("Should add wallet1 as auth", async function () {
+      const spells = {
+        connector: "authV2",
+        method: "add",
+        args: [wallet1.address]
+      }
+      const tx = await acountV2DsaM1Wallet0.connect(wallet0).cast(...encodeSpells([spells]), wallet1.address)
+      const receipt = await tx.wait()
+      const logCastEvent = expectEvent(receipt, (await deployments.getArtifact("InstaAccountV2ImplementationM1")).abi, "LogCast")
+      const LogAddAuthEvent = expectEvent(receipt, (await deployments.getArtifact("ConnectV2Auth")).abi, "LogAddAuth")
+    });
+
+    it("Should add wallet2 as auth", async function () {
+      const spells = {
+        connector: "authV2",
+        method: "add",
+        args: [wallet2.address]
+      }
+      const tx = await acountV2DsaM2Wallet0.connect(wallet1).castWithFlashloan(...encodeSpells([spells]), wallet1.address)
+      const receipt = await tx.wait()
+      const logCastEvent = expectEvent(receipt, (await deployments.getArtifact("InstaAccountV2ImplementationM2")).abi, "LogCast")
+      const LogAddAuthEvent = expectEvent(receipt, (await deployments.getArtifact("ConnectV2Auth")).abi, "LogAddAuth")
+    });
+
+    it("Should remove wallet1 as auth", async function () {
+      const spells = {
+        connector: "authV2",
+        method: "remove",
+        args: [wallet1.address]
+      }
+      const tx = await acountV2DsaM1Wallet0.connect(wallet2).cast(...encodeSpells([spells]), wallet2.address)
+      const receipt = await tx.wait()
+      expectEvent(receipt, (await deployments.getArtifact("InstaAccountV2ImplementationM2")).abi, "LogCast")
+      expectEvent(receipt, (await deployments.getArtifact("ConnectV2Auth")).abi, "LogRemoveAuth")
+    });
+
+    // This one fails
+    // it("Should add wallet3 as auth using default implmentation", async function() {
+    //   console.log("uwu", "msg.sender=", wallet0.address, "address(this)=" ,acountV2DsaDefaultWallet0.address)
+    //   const tx = await acountV2DsaDefaultWallet0.connect(wallet0).enable(wallet3.address)
+    //   const receipt = await tx.wait()
```

Comment:
@thrilok209 This fails for some reason. Everything else is passing (except for Compound withdraw, which is failing in the cToken contract)